tests/unit/iontests.js

```diff
@@ -11,126 +11,365 @@
+                    resolve();
+                };
+
+                return new Promise(executor);
+            }
+        }
+
+        function makeEventStreamTest(path) {
+            return function() {
+                let executor = function(resolve, reject) {
+                    let options = path.endsWith(".10n") ? null : "utf8";
+                    let input = fs.readFileSync(path, options);
+                    roundTripEventStreams(ion.makeReader(input));
+                    resolve();
+                };
+
+                return new Promise(executor);
+            }
+        }
+
+        function makeBadEventStreamTest(path) {
+            return function() {
+                let executor = function(resolve, reject) {
+                    let options = path.endsWith(".10n") ? null : "utf8";
+                    let input = fs.readFileSync(path, options);
+                    roundTripBadEventStreams(ion.makeReader(input));
+                    resolve();
+                };
+
+                return new Promise(executor);
+            }
+        }
+
+        function roundTripEventStreams(reader){
+            let eventStream = new ion.IonEventStream(reader);
+            let writer = ion.makeTextWriter();
+            eventStream.writeEventStream(writer);
+            writer.close();
+            let buf = writer.getBytes();
+            let tempString = '';
+            for(let i = 0; i < buf.length; i++){
+                tempString = tempString + String.fromCharCode(buf[i]);
+            }
+            let tempReader = new ion.makeReader(tempString);
+            let tempStream = new ion.IonEventStream(tempReader);
+            if(!eventStream.equals(tempStream)) {
+                let tempWriter = ion.makeTextWriter();
+                tempStream.write(tempWriter);
+                tempWriter.close();
+                let tempBuf = tempWriter.getBytes();
+                let unequalString = "";
+                for(let i = 0; i < buf.length; i++){
+                    unequalString = unequalString + String.fromCharCode(buf[i]);
+                }
+                throw new Error('Round tripped stream was unequal: ' + tempString + '\n vs: ' + unequalString);
+            }
+
+
+        }
+
+        function roundTripBadEventStreams(reader){
             try {
-              if (typeof(chunk) !== 'undefined') {
-                chunks.push(chunk);
-              }
-              var buffer = Buffer.concat(chunks);
-              var reader = ion.makeReader(buffer);
-              console.log("Exhausting " + path);
-              if (path.endsWith('clobsWithWhitespace.ion')) {
-              }
-              exhaust(reader);
-              resolve();
-            } catch (e) {
-              reject(e);
-            }
-          });
-
-          stream.on('data', function(chunk) {
-            chunks.push(chunk);
-          });
-        };
+                let eventStream = new ion.IonEventStream(reader);
```

Comment:
All these files should fail on read, which happens here. There's nothing to re-write.